make_grammar.js

```diff
@@ -262,8 +262,8 @@ module.exports = function make_grammar(dialect) {
       },
 
       template_expr: $ => choice(
-        $.quoted_template,
-        $.heredoc_template,
+        field('quotedTemplate', $.quoted_template),
```

Comment:
For `choice`es it is normally best to not use field names, it would result in classes like
```ql
class TemplateExpr {
  /** Get a quoted templated if this TemplateExpr is a QuoteTemplate */
  QuotedTemplate getQuotedTemplate();
  /** Get a quoted templated if this TemplateExpr is a HeredocTemplate */
  HeredocTemplate getHeredocTemplate();
}

Without the field names we should end up with something like. It might need some `superTypes` property in the grammar to get the inheritance right.
```ql
class TemplateExpr { ... }
class QuotedTemplate extends TemplateExpr { ... }
class HeredocTemplate extends TemplateExpr { ... }
```